syntax = "proto3";
package stockbuddy;

message Quote {
	string symbol = 1;
	int32 timestamp = 2;
	float open = 3;
	float high = 4;
	float low = 5;
	float close = 6;
	float adj_close = 7;
	int32 volume = 8;
}

message QuoteRequest {
	string symbol = 1;
	int32 period = 2;
}

message QuoteResponse {
	repeated Quote quotes = 1;
}

service QuoteService {
	rpc ListQuoteHistory(QuoteRequest) returns (QuoteResponse);
}
